File: database/src/main/java/com/clpstudio/domainlib/dagger/DiComponent.java
package com.clpstudio.domainlib.dagger;

import javax.inject.Singleton;

import dagger.Component;

/**
 * Created by clapalucian on 03/06/2017.
 */

@Component(
        modules = {
                LibraryModule.class,
                RetrofitModule.class,
                FirebaseModule.class
        }
)
@Singleton
public interface DiComponent {

}

File: app/src/main/java/com/clpstudio/bsocial/core/dagger/DIComponent.kt
package com.clpstudio.bsocial.core.dagger

import com.clpstudio.bsocial.presentation.MainActivity
import dagger.Component
import javax.inject.Singleton

@Component(modules = arrayOf(ApplicationModule::class))
@Singleton
interface DIComponent {
    // AutoFactory classes used in modules
    //    ApiClientFactory provideApiClientFactory();

    // injects
    fun inject(inject: MainActivity): MainActivity
}
